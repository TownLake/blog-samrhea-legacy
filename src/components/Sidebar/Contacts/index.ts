export { default as Contacts } from "./Contacts";
