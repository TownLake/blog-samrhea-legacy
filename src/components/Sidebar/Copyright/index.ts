export { default as Copyright } from "./Copyright";
