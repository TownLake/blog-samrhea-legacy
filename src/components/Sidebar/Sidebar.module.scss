@import "../../assets/scss/mixins";

.sidebar {
  width: 100%;

  .inner {
    @include padding(1, 0.75, 0);
    position: relative;

    &:after {
      background: var(--color-sidebar-border);
      background: linear-gradient(
          to bottom,
          var(--color-sidebar-border) 0%,
          var(--color-sidebar-border) 48%,
          var(--color-sidebar-border-fade) 100%
      );
    }
  }
}

@include breakpoint-sm {
  .sidebar {
    lost-column: 5/12;

    .inner {
      @include padding(1.25, 0.75, 0);

      &:after {
        bottom: 0;
        content: "";
        height: 540px;
        position: absolute;
        right: -10px;
        top: 30px;
        width: 1px;
      }
    }
  }
}

@include breakpoint-md {
  .sidebar {
    lost-column: 1/3;

    .inner {
      @include padding-equal(1.5);
    }
  }
}
