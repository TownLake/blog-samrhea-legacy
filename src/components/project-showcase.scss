.project-showcase {
    max-width: 1200px;
    margin: 0 auto;
    padding: 2rem;
  
    h1 {
      text-align: center;
      margin-bottom: 2rem;
      font-size: 2rem;
      font-weight: bold;
    }
  
    .project-grid {
      display: grid;
      grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
      gap: 2rem;
    }
  }
  
  // Dark mode styles
  @media (prefers-color-scheme: dark) {
    .project-showcase {
      background-color: #1a202c;
      color: #f7fafc;
    }
  }