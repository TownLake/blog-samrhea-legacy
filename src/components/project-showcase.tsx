.project-tile {
    position: relative;
    width: 100%;
    height: 0;
    padding-bottom: 75%; // 4:3 aspect ratio
    overflow: hidden;
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
  
    &:hover {
      transform: translateY(-5px);
    }
  
    .project-link {
      text-decoration: none;
      color: inherit;
    }
  
    .project-image {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-size: cover;
      background-position: center;
    }
  
    .project-overlay {
      position: absolute;
      bottom: 0;
      left: 0;
      right: 0;
      background: rgba(0, 0, 0, 0.7);
      color: white;
      padding: 1rem;
      transform: translateY(100%);
      transition: transform 0.3s ease;
    }
  
    &:hover .project-overlay {
      transform: translateY(0);
    }
  
    .project-title {
      margin: 0 0 0.5rem;
      font-size: 1.2rem;
    }
  
    .project-description {
      margin: 0;
      font-size: 0.9rem;
      opacity: 0.8;
    }
  }