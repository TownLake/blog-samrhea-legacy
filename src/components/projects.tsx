import React from 'react';
import { Link } from 'gatsby';
import './project.scss';

interface ProjectProps {
  title: string;
  description: string;
  imageUrl: string;
  projectUrl: string;
}

const Project: React.FC<ProjectProps> = ({ title, description, imageUrl, projectUrl }) => {
  return (
    <div className="project-tile">
      <Link to={projectUrl} className="project-link">
        <div className="project-image" style={{ backgroundImage: `url(${imageUrl})` }}>
          <div className="project-overlay">
            <h3 className="project-title">{title}</h3>
            <p className="project-description">{description}</p>
          </div>
        </div>
      </Link>
    </div>
  );
};

export default Project;