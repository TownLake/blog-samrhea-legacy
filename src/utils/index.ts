export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
