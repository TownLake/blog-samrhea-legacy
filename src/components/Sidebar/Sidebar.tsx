import React from "react";

import { useSiteMetadata } from "@/hooks";

import { Author } from "./Author";
import { Contacts } from "./Contacts";
import { Copyright } from "./Copyright";
import { Menu } from "./Menu";

import * as styles from "./Sidebar.module.scss";

type Props = {
  isIndex?: boolean;
};

const Sidebar = ({ isIndex }: Props) => {
  const { author, copyright, menu } = useSiteMetadata();

  return (
    <div className={styles.sidebar}>
      <div className={styles.inner}>
        <Author author={author} isIndex={isIndex} />
        <Menu menu={menu} />
        <Contacts contacts={author.contacts} />
        <Copyright copyright={copyright} />
      </div>
    </div>
  );
};

export default Sidebar;
