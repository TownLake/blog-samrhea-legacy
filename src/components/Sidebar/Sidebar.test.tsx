import React from "react";

import { StaticQuery, useStaticQuery } from "gatsby";

import { Sidebar } from "@/components/Sidebar";
import * as mocks from "@/mocks";
import { testUtils } from "@/utils";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("Sidebar", () => {
  beforeEach(() => {
    mockedStaticQuery.mockImplementationOnce(({ render }) =>
      render(mocks.siteMetadata),
    );
    mockedUseStaticQuery.mockReturnValue(mocks.siteMetadata);
  });

  test("renders correctly", () => {
    const props = { isIndex: true };
    const tree = testUtils
      .createSnapshotsRenderer(<Sidebar {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
