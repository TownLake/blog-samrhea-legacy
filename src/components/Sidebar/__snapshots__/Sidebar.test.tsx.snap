// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div>
  <div>
    <div>
      <a
        href="/"
      />
      <div>
        <h1>
          <a
            href="/"
          >
            John Doe
          </a>
        </h1>
      </div>
      <p>
        Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.
      </p>
    </div>
    <nav>
      <ul>
        <li>
          <a
            href="/"
          >
            Articles
          </a>
        </li>
        <li>
          <a
            href="/pages/about"
          >
            About Me
          </a>
        </li>
        <li>
          <a
            href="/pages/contacts"
          >
            Contact Me
          </a>
        </li>
      </ul>
    </nav>
    <div>
      All rights reserved.
    </div>
  </div>
</div>
`;
